### DISCUSSION TITLE
A K3S cluster working in a dual network card  environment is in doubt

### DISCUSSION THREAD

**Discussion by ddddage** [k3s-io/k3s#6564] [Q&A] [answered]
I have a cluster of virtual machines, each virtual machine has two NICs, one of which is used to communicate between virtual machines and the other is used to communicate with the Internet. An accident occurred when I wanted to connect to the master node with the agent node: when connecting to the master with the agent, the log shows that two IPs were requested, but my configuration only uses one IP, because the other IP is not available between VMs, does anyone know how to remove one of the IPs
![image](https://user-images.githubusercontent.com/25663972/204137573-0bc44e63-a49d-442b-8724-dd0de1171416.png)
In the environment shown in the picture, I just want to connect 192.168.117.11:6443 instead of connecting two
thanks

**Comment by brandond** [comment, marked as answer]
One of the IPs is the address you specified for the --server address. The other is the address that the server node is using within the cluster. If you want to override that, you should start both the server and the agent with --node-ip set to the address of the interface you prefer to use.

I'm not sure what you mean by "dual graphics". What is that about?

**Reply by ddddage** [reply]
There were some mistakes I described earlier. The problem I described was the connection error when using a virtual machine with two network cards In the picture, the IP address 10.0.2. x is set as the nat configuration for connecting to the external network. The IP address 192.168.117.xx is the configuration for communication between virtual machines. Theoretically, the 10.0.2. x network segment should not be used for any configuration, because it can only connect to the external network, and it is invalid for virtual machines to communicate with each other. I have just started using k3s recently, and my understanding of the configuration is not very comprehensive. However, I think there will be some problems with the 10.0.2. x network segment in the log

**Reply by ddddage** [reply]
According to your answer, I think I should specify the "-- node ip" configuration. But is it correct to configure the "node ip" to the network segment of another existing network card when I do not specify it, or is it just a coincidence? And thank you for your answer

**Reply by brandond** [reply]
I'm not sure exactly what you're asking, but as I said above you can use that flag to set the IP that you would prefer that node to use as it's private address, which determines how other nodes will connect to it. By default it uses the IP with the default route (the one that can be used to reach the internet).
